### PR TITLE
[FIX] resourceListCreator: Include dependencies of subModules

### PR DESCRIPTION
**Pull request by matz3**
JIRA: CPOUI5FOUNDATION-310

### CODE REVIEW

File: lib/lbt/resources/ResourceCollector.js [hunk 1]
```diff
@@ -109,29 +109,40 @@ class ResourceCollector {
...
+					// Add subModule as included
+					resourceInfo.included.add(subModule);
+
+					// Try to inherit dependency info
+					let subModuleInfo;
+					try {
+						subModuleInfo = await this._pool.getModuleInfo(subModule);
+					} catch (err) {
+						// TODO: Decide on message and log level
+						log.warn("Missing module? " + subModule);
```
Comment (matz3) [inline lib/lbt/resources/ResourceCollector.js:134]: verbose

Comment (matz3) [reply lib/lbt/resources/ResourceCollector.js]: Done

File: lib/lbt/resources/ResourceCollector.js [hunk 2]
```diff
@@ -109,29 +109,39 @@ class ResourceCollector {
...
+					// Try to inherit dependency info
+					let subModuleInfo;
+					try {
+						subModuleInfo = await this._pool.getModuleInfo(subModule);
+					} catch (err) {
+						log.verbose(`	missing submodule ${subModule} included by ${moduleInfo.name}`);
+					}
+					if (subModuleInfo) {
+						// Inherit dependencies
+						this._applyModuleInfoDependenciesToResourceInfo(resourceInfo, subModuleInfo);
```
Comment (matz3) [inline lib/lbt/resources/ResourceCollector.js:137]: Dependencies should not be added if they are already listed as "included". I'm working on a fix..

Comment (matz3) [reply lib/lbt/resources/ResourceCollector.js:137]: Also, a dependency should not be listed as required **and** condRequired

**Comment by matz3** [comment]
Ready for review

**Comment by RandomByte** [comment]
LGTM. But @codeworrior should definitely take a look at the dependency inheritance logic

**Review by codeworrior — APPROVED** [review]
It's a bit unfortunate that we have to repeat the dependency logic from ModuleInfo here, but ResourceInfo was kept rather simple (no logic, just a JSON-like object).

I'm also still struggling with the fact that eager dependencies of submodules become eager dependencies of the bundle (what if a submodule is not used at runtime?). Not sure whether this was a concrete requirement of the HAT packager or whether this behavior of Maven was "by accident".

But as the current focus is to get rid of differences between Maven and ui5-tooling, Im fine with the change.

**Comment by matz3** [comment]
Thanks for the feedback! I agree. We should right now only focus on having the same behavior. Changes / improvements can then still be made later on if required.

**Pull request merged**